### ISSUE TITLE
[question] Usage of conan_build_info --v2

### ISSUE THREAD

**Issue by djeverett** [conan-io/conan#8342]
I have tried using `conan_build_info --v2 create --lockfile=<lockfile> build_info.json` to create build info.

The result:
```
{
    "version": "1.0.1",
    "name": "forge :: users :: daneve01 :: forge_libb",
    "number": "1",
    "type": "GENERIC",
    "started": "2021-01-15T15:57:10.000Z",
    "buildAgent": {
        "name": "Conan Client",
        "version": "1.X"
    },
    "modules": []
}
```
Is it really right that `modules` is not populated?  Am I using this correctly?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @djeverett 

No it shouldn't, there should be something inside. Can you please check the ``lockfile`` that you are using as argument? Can you share it?

**Comment by djeverett** [comment]
The lockfile is:

```
{
 "graph_lock": {
  "nodes": {
   "0": {
    "ref": "forge_libb/0.1@forge/testing#bea7c7f83a31f06d75062276090f60c2",
    "requires": [
     "1"
    ],
    "context": "host"
   },
   "1": {
    "ref": "forge_liba/0.1@forge/testing#d843a50d74181a941f2ed7029ddd7b4a",
    "package_id": "b80d46004713aa37d6a90b42e2a326a056a237b5",
    "prev": "8b4e26468f15b4be9361c97d04c5f63a",
    "context": "host"
   }
  },
  "revisions_enabled": true
 },
 "version": "0.4",
 "profile_host": "[settings]\narch=x86_64\narch_build=x86_64\nbuild_type=Release\ncompiler=gcc\ncompiler.libcxx=libstdc++11\ncompiler.version=8\nos=Linux\nos_build=Linux\n[options]\n[build_requires]\n[env]\n"
}
```
I am using Conan 1.32.0.

**Comment by czoido** [comment]
Hi @djeverett,
I think that the build info you are getting is empty because when you executed the command that used the lockfile you did not build anything or you did not get an output lockfile using the `--lockfile-out` argument. 
If you do so there should appear a field named `modified` set to value `true`. **Only the nodes marked as modified** will be included in the build info file. Something like this:

```
   "3": {
    "ref": "liba/1.0#0b1a78f5925f62480ed2d97002e926fa",
    "options": "fPIC=True\nshared=False",
    "package_id": "6a83d7f783e7ee89a83cf2fe72b5f5f67538e2a6",
    "prev": "5a30794dc739189b028d191812c2b845",
    "modified": true,
    "context": "host"
   },
```

Can you check the steps followed to get the lockfile and if you are really building something in the process or if you are not updating the lock, something like:

```
conan install . --lockfile=conan.lock --lockfile-out=updated.lock --build
```

Here you can find a minimal example on how to generate a build info from a lockfile: https://github.com/czoido/conan-build-info-example 

Hope this helps, please tell me if you manage to generate the build info correctly.

**Comment by djeverett** [comment]
Ah, I was following the stages in [the CI guide](https://docs.conan.io/en/latest/versioning/lockfiles/ci.html).  Looks like I need to use `conan install --build --lockfile-out=<lockfile-out>` rather than `conan create`.  Will test and get back to you.

**Comment by czoido** [comment]
Hi @djeverett,
It's ok if you use conan `create` but always check that you build something and generate an updated lockfile using `--lockfile-out` you could also use in the example `conan create . --lockfile=conan.lock --lockfile-out=updated.lock --build`

**Comment by djeverett** [comment]
Thanks @czoido  - that all seems to work.  

There remains one outstanding issue: the [conan create](https://docs.conan.io/en/latest/reference/commands/creator/create.html) documentation does not document the `--lockfile-out` argument (`conan create --help` does however).

**Comment by czoido** [comment]
> Thanks @czoido - that all seems to work.
> 
Great! Glad to hear that
> There remains one outstanding issue: the [conan create](https://docs.conan.io/en/latest/reference/commands/creator/create.html) documentation does not document the `--lockfile-out` argument (`conan create --help` does however).

Yes, sorry for that, we have already corrected the documentation for 1.33.0 release that will be released today.

**Comment by czoido** [comment]
I'm closing this issue, please feel free to reopen if you have any more questions.
